tritium-core/src/main/java/com/palantir/tritium/event/CompositeInvocationEventHandler.java

```diff
@@ -32,10 +33,11 @@
 
     private static final Logger logger = LoggerFactory.getLogger(CompositeInvocationEventHandler.class);
 
-    private final List<InvocationEventHandler<InvocationContext>> handlers;
+    private final InvocationEventHandler<InvocationContext>[] handlers;
 
+    @SuppressWarnings("unchecked")
     private CompositeInvocationEventHandler(List<InvocationEventHandler<InvocationContext>> handlers) {
-        this.handlers = ImmutableList.copyOf(handlers);
+        this.handlers = ImmutableList.copyOf(handlers).toArray(new InvocationEventHandler[0]);
```

Comment:
should we remove the `ImmutableList` copy?

```suggestion
        this.handlers = handlers.toArray(new InvocationEventHandler[0]);
```

Reply:
Good call, it's an expensive null element check, I'll write a loop.